src/machine.rs

```diff
@@ -998,6 +998,7 @@ impl OlmMachine {
 
         // convert the js-side data into rust data
         let mut keys: BTreeMap<_, BTreeMap<_, _>> = BTreeMap::new();
+        let mut failures: usize = 0;
```

Comment:
I don't know if this is more idiomatic:

```suggestion
        let mut failures = 0usize;
```

Reply:
Can't the compiler infer it should be a usize? If not, can we use `let mut failures = 0` here (assuming i32::max_value() is sufficient for the max number of failures) and then the one user can be replaced with `JsValue::from(failures as usize)`?